tests/pyfunc/test_spark.py

```diff
@@ -67,6 +66,16 @@ def score_model_as_udf(model_uri, pandas_df, result_type="double"):
     return [x["prediction"] for x in new_df.collect()]
 
 
+def score_model_as_udf(model_uri, pandas_df, result_type="double"):
+    if spark := pyspark.sql.SparkSession.getActiveSession():
+        # Reuse the active SparkSession, don't kill it after use
+        return score_spark(spark, model_uri, pandas_df, result_type)
```

Comment:
Do we need this? `get_spark_session` returns `getOrCreate`.
If there's no spark session, with below context manager, we'll never reach here right? If we want to reuse the same spark session, we might need to unittest and clear within `teardown` 🤔

Reply:
This function is imported and used in different test modules.

Reply:
We could add `spark` argument to `score_model_as_udf`.

Reply:
I mean we could just use `with get_spark_session(pyspark.SparkConf()) as spark:`. If there's no spark session it will create one, if there is it just fetches it.

Reply:
```
def score_model_as_udf(model_uri, pandas_df, result_type="double"):
    with get_spark_session(pyspark.SparkConf()) as spark:
        return score_spark(spark, model_uri, pandas_df, result_type)
```

Or if we could reuse the `spark` fixture, we can directly use `score_spark` function instead of this?

Reply:
```
def score_model_as_udf(model_uri, pandas_df, result_type="double"):
    with get_spark_session(pyspark.SparkConf()) as spark:
        return score_spark(spark, model_uri, pandas_df, result_type)
```

is going to kill the active spark, right?

Reply:
Yes

Reply:
I see, this is for spark session that's created outside of this function.

Reply:
Yep

Reply:
But looks like we don't have such usecases in our tests. BTW some places import this functino but then defines it again, e.g. test_spark_connect_model_export.py

Reply:
Then I think this PR is good to go! But we might want to refactor those spark tests afterwards.